File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r= height.length-1;
        int maxL = 0;
        while(l<r){
                  int length = Math.min(height[r],height[l]);
            int breadth = r-l;
            int area = length * breadth;
            maxL = Math.max(area,maxL);
            if(height[l] < height[r]){
          
l++;
            }else{
                r--;
            }
        }
        return maxL;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l = 0
        r =len(height)-1
        maxL = 0
        while  l < r:
            area = r-l
            maxL = max(maxL,area * min(height[r],height[l]))
            if height[l] <= height[r]:
                
                l +=1
            else:
                r-=1
        return maxL
